### ISSUE TITLE
[Paimon Catalog] Querying data from primary key table raise an error

### ISSUE THREAD

**Issue by 0akarma** [StarRocks/starrocks#30674]
### Steps to reproduce the behavior (Required)
```
-- DROP TABLE row_error_test;
CREATE TABLE IF NOT EXISTS row_error_test (
    f_random_str STRING NOT NULL
);
-- DROP TEMPORARY TABLE datagen_source;
CREATE TEMPORARY TABLE datagen_source (
    f_random_str STRING,
    PRIMARY KEY(f_random_str) NOT ENFORCED
) WITH (
 'connector' = 'datagen',
 'number-of-rows' = '10',
 'fields.f_random_str.length'='100'
);


INSERT INTO row_error_test select * from datagen_source;
```

### Expected behavior (Required)
Flink SQL return resut below 
```
Flink SQL> select * from row_error_test limit 1;
2023-09-08 22:28:16,739 INFO  org.apache.hadoop.fs.s3a.S3AInputStream                      [] - Switching to Random IO seek policy
2023-09-08 22:28:16,803 INFO  org.apache.hadoop.fs.s3a.S3AInputStream                      [] - Switching to Random IO seek policy
2023-09-08 22:28:16,822 INFO  org.apache.hadoop.fs.s3a.S3AInputStream                      [] - Switching to Random IO seek policy
+--------------------------------+
|                   f_random_str |
+--------------------------------+
| e0d61319098f682e052c62c0b6f... |
+--------------------------------+
1 row in set
```
### Real behavior (Required)
Sql-client return an error below
```
StarRocks > select * from row_error_test limit 1;
ERROR 1064 (HY000): All columns must be nullable for external table. Column 'f_random_str' is not nullable, You can rebuild theexternal table and We strongly recommend that you use catalog to access external data.

```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
in branch `main`  and the latest commit is da50131

**Comment by leoyy0316** [comment]
fixed by #30716

**Comment by 0akarma** [comment]
LGTM
